string agent
string command
string type
float64[] gesture
float64[] gps
